File: Ruby/fizzbuzz.rb
# naive implementation
1.upto (2**64 -1) do |i|
  str = ""
  if i % 3 == 0
    str += "fizz"
  end
  if i % 5 == 0
    str += "buzz"
  end
  if str == ""
    str += i.to_s
  end
  puts str
end

File: C/fizzbuzz.c
/*
 * naive implementation
 */

#include <stdio.h>
#include <limits.h>
#include <string.h>

void main() {
    char out[20];

    for (int i = 1; i < INT_MAX; i++) {
        out[0] = '\0';
        if (i % 3 == 0) {
            strcat(out, "fizz");
        }
        if (i % 5 == 0) {
            strcat(out, "buzz");
        }
        if (strcmp(out, "") == 0) {
            sprintf(out, "%d", i);
        }
        printf("%s\n", out);
    }
}
